src/modules/hip/kernel/fog.hpp

```diff
@@ -0,0 +1,323 @@
+#include <random>
+#include <hip/hip_runtime.h>
+
+#include "rpp_hip_common.hpp"
+#include "fog_mask.hpp"
+
+__device__ __forceinline__ void fog_hip_compute(uchar *srcPtr, d_float8 *src_f8, d_float8 *dst_f8, d_float8 *maskAlpha_f8, d_float8 *maskIntensity_f8, float4 *gamma_f4)
+{
+    float4 alphaFactor_f4[2];
+    alphaFactor_f4[0] = rpp_hip_pixel_check_0to1(maskAlpha_f8->f4[0] + *gamma_f4);
```

Comment:
why rpp_hip_pixel_check_0to1 is used here?

Reply:
Since we adding gamma to alpha value There may be chance to get alpha value beyond the range of (0-1) .So we need to check the alpha value is in range or not .